api/providers/tasks.py

```diff
@@ -486,9 +487,16 @@ def handle_registration_row(row, initiator, provider, schema, auto_approval=Fals
             error = 'Initiator [{}] is not affiliated with institution [{}]'.format(initiator._id, institution._id)
             raise RegistrationBulkCreationRowError(row.upload.id, row.id, row_title, row_external_id, error=error)
 
-    # Set registration responses
-    # TODO: if available, capture specific exceptions during setting responses
-    draft.update_registration_responses(responses)
+    # Validate and set registration responses
+    try:
+        # TODO: figure out why validation fails when `required_fields=True`
+        draft.validate_registration_responses(responses, required_fields=False)
```

Comment:
First, `draft.validate_metadata()` fails as expected since the API code only log the exception.

In addition, `draft.validate_registration_responses(responses, required_fields=False)` actually caught an issue I had in the CSV template (empty answer to Q27).

However, `draft.validate_registration_responses(responses, required_fields=True)` fails even I have every cell populated.

Reply:
So what is the ValidationError thrown when `required_fields=True`?

Reply:
As discussed, removed.